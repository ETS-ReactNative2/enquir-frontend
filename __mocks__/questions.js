const questions = [
    {
        id: 0,
        questionInput: 'If mirrors aren\'t real, how can mirrors be real?',
    },
];

module.exports = {
    questions
};