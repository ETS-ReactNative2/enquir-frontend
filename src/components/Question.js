import React, { Component } from 'react';
import { Modal, Text, FlatList, Button, View, StyleSheet, TextInput, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

export default class Question extends Component {
    state = {
        question: this.props.question,
        editing: false,
    };

    edit = () => {
        this.setState({
            editing: !this.state.editing,
        });
    };

    render() {
        const { question, editing } = this.state;

        return (
            <View style={styles.questionContainer}>
                <TextInput
                    style={styles.questionText}
                    value={question.item}
                    editable={editing}
                />
                <View style={styles.questionEditContainer}>
                    <TouchableOpacity onPress={this.edit}>
                        <View style={styles.editQuestion}>
                            <Icon name='pencil' size={20} />
                        </View>
                    </TouchableOpacity>
                    <TouchableOpacity onPress={() => this.props.delete(question)}>
                        <View style={styles.deleteQuestion}>
                            <Icon name='trash' size={20} />
                        </View>
                    </TouchableOpacity>
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    questionContainer: {
        flexDirection: 'row',  // main axis
        justifyContent: 'space-between', // main axis
        paddingTop: 10,
        paddingBottom: 10,
        paddingLeft: 18,
        paddingRight: 16,
        marginTop: 0,
        marginBottom: 8,
        elevation: 1,
        borderRadius: 3,
        backgroundColor: 'grey',
    },
    questionText: {
        justifyContent: 'flex-start',
    },
    questionEditContainer: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'flex-end',
    },
    editQuestion: {
        marginRight: 12,
    },
    deleteQuestion: {},
});