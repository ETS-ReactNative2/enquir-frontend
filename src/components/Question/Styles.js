import { StyleSheet } from 'react-native';

const containers = StyleSheet.create({
    questionContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        paddingTop: 10,
        paddingBottom: 10,
        paddingLeft: 18,
        paddingRight: 16,
        marginTop: 0,
        marginBottom: 8,
        elevation: 1,
        borderRadius: 3,
        backgroundColor: 'grey',
    },
    questionTextContainer: {
        flex: 12,
        flexDirection: 'row',
        justifyContent: 'flex-start',
    },
    editMenu: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'flex-end',
    },
});

const text = StyleSheet.create({
    questionText: {
        flex: 1,
        flexWrap: 'wrap',
        fontSize: 16,
        fontFamily: 'Roboto-Regular',
    },
});

export { containers, text };