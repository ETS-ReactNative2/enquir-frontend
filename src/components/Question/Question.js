import React, { Component } from 'react';
import { 
    View, 
    TextInput, 
    TouchableOpacity,
} from 'react-native';
import { containers, text } from './Styles';
import Icon from 'react-native-vector-icons/FontAwesome';

export default class Question extends Component {
    state = {
        question: this.props.question,
        questionInput: this.props.question.item,
    };

    render() {
        const { question, questionInput } = this.state;

        return (
            <View style={containers.questionContainer}>
                <View style={containers.questionTextContainer}>
                    <TextInput
                        style={text.questionText}
                        onChangeText={(questionInput) => this.setState({ questionInput })}
                        onEndEditing={() => this.props.save(question.index, questionInput)}
                        value={questionInput}
                        multiline={true}
                    />
                </View>
                <View style={containers.editMenu}>
                    <TouchableOpacity onPress={() => this.props.delete(question.index)}>
                        <View>
                            <Icon name='trash' size={18} />
                        </View>
                    </TouchableOpacity>
                </View>
            </View>
        );
    }
};