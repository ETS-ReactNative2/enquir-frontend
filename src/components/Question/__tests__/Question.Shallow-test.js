import React from 'react';
import { TextInput } from 'react-native';
import { mount } from 'enzyme';

import Question from '../Question';
import { questions } from '../../../../__mocks__/questions';

const question = questions[0];

test('renders a Question using Enzyme 1', () => {
  const wrapper = mount(
    <Question
      id={question.id}
      question={''}
    />
  );
  expect(wrapper).toMatchSnapshot();
  expect((wrapper).find(TextInput).props().value.length).toEqual(0);
});

test('renders a Question using Enzyme 2', () => {
  const wrapper = mount(
    <Question
      id={question.id}
      question={question.questionInput}
    />
  );
  expect(wrapper).toMatchSnapshot();
  expect((wrapper).find(TextInput).props().value.length).toBeGreaterThan(0);
});