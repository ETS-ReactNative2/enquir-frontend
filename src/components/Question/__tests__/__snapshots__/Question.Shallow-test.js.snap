// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Question using Enzyme 1 1`] = `
<Question
  id={0}
  question=""
>
  <View
    style={
      Object {
        "backgroundColor": "#c3c4c9",
        "borderRadius": 3,
        "elevation": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 8,
        "marginTop": 0,
        "paddingBottom": 10,
        "paddingLeft": 18,
        "paddingRight": 16,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#c3c4c9",
          "borderRadius": 3,
          "elevation": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 8,
          "marginTop": 0,
          "paddingBottom": 10,
          "paddingLeft": 18,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 12,
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 12,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <Component
            allowFontScaling={true}
            multiline={true}
            onChangeText={[Function]}
            onEndEditing={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "flex": 1,
                "flexWrap": "wrap",
                "fontFamily": "Roboto-Regular",
                "fontSize": 18,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          >
            <TextInput
              allowFontScaling={true}
              multiline={true}
              onChangeText={[Function]}
              onEndEditing={[Function]}
              rejectResponderTermination={true}
              style={
                Object {
                  "flex": 1,
                  "flexWrap": "wrap",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 18,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </Component>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
        >
          <ForwardRef
            onPress={[Function]}
          >
            <TouchableOpacity
              hostRef={null}
              onPress={[Function]}
            >
              <ForwardRef(AnimatedComponentWrapper)
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    undefined,
                    Object {
                      "opacity": 1,
                    },
                  ]
                }
              >
                <AnimatedComponent
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      undefined,
                      Object {
                        "opacity": 1,
                      },
                    ]
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View>
                        <View>
                          <Icon
                            allowFontScaling={false}
                            name="trash"
                            size={18}
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": undefined,
                                    "fontSize": 18,
                                  },
                                  undefined,
                                  Object {
                                    "fontFamily": "FontAwesome",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                  Object {},
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": undefined,
                                      "fontSize": 18,
                                    },
                                    undefined,
                                    Object {
                                      "fontFamily": "FontAwesome",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                    Object {},
                                  ]
                                }
                              >
                                ï‡¸
                              </Text>
                            </Text>
                          </Icon>
                        </View>
                      </View>
                      <PressabilityDebugView
                        color="cyan"
                      />
                    </View>
                  </View>
                </AnimatedComponent>
              </ForwardRef(AnimatedComponentWrapper)>
            </TouchableOpacity>
          </ForwardRef>
        </View>
      </View>
    </View>
  </View>
</Question>
`;

exports[`renders a Question using Enzyme 2 1`] = `
<Question
  id={0}
  question="If mirrors aren't real, how can mirrors be real?"
>
  <View
    style={
      Object {
        "backgroundColor": "#c3c4c9",
        "borderRadius": 3,
        "elevation": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 8,
        "marginTop": 0,
        "paddingBottom": 10,
        "paddingLeft": 18,
        "paddingRight": 16,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#c3c4c9",
          "borderRadius": 3,
          "elevation": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 8,
          "marginTop": 0,
          "paddingBottom": 10,
          "paddingLeft": 18,
          "paddingRight": 16,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 12,
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 12,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <Component
            allowFontScaling={true}
            multiline={true}
            onChangeText={[Function]}
            onEndEditing={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "flex": 1,
                "flexWrap": "wrap",
                "fontFamily": "Roboto-Regular",
                "fontSize": 18,
              }
            }
            underlineColorAndroid="transparent"
            value="If mirrors aren't real, how can mirrors be real?"
          >
            <TextInput
              allowFontScaling={true}
              multiline={true}
              onChangeText={[Function]}
              onEndEditing={[Function]}
              rejectResponderTermination={true}
              style={
                Object {
                  "flex": 1,
                  "flexWrap": "wrap",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 18,
                }
              }
              underlineColorAndroid="transparent"
              value="If mirrors aren't real, how can mirrors be real?"
            />
          </Component>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
        >
          <ForwardRef
            onPress={[Function]}
          >
            <TouchableOpacity
              hostRef={null}
              onPress={[Function]}
            >
              <ForwardRef(AnimatedComponentWrapper)
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    undefined,
                    Object {
                      "opacity": 1,
                    },
                  ]
                }
              >
                <AnimatedComponent
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      undefined,
                      Object {
                        "opacity": 1,
                      },
                    ]
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View>
                        <View>
                          <Icon
                            allowFontScaling={false}
                            name="trash"
                            size={18}
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": undefined,
                                    "fontSize": 18,
                                  },
                                  undefined,
                                  Object {
                                    "fontFamily": "FontAwesome",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                  Object {},
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": undefined,
                                      "fontSize": 18,
                                    },
                                    undefined,
                                    Object {
                                      "fontFamily": "FontAwesome",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                    Object {},
                                  ]
                                }
                              >
                                ï‡¸
                              </Text>
                            </Text>
                          </Icon>
                        </View>
                      </View>
                      <PressabilityDebugView
                        color="cyan"
                      />
                    </View>
                  </View>
                </AnimatedComponent>
              </ForwardRef(AnimatedComponentWrapper)>
            </TouchableOpacity>
          </ForwardRef>
        </View>
      </View>
    </View>
  </View>
</Question>
`;
